crates/oxc_language_server/src/main.rs

```diff
@@ -492,21 +544,36 @@ impl Backend {
         if config.exists() {
             config_path = Some(config);
         }
-        if let Some(config_path) = config_path {
-            let mut linter = self.server_linter.write().await;
-            let config = Oxlintrc::from_file(&config_path)
-                .expect("should have initialized linter with new options");
-            let config_store = ConfigStoreBuilder::from_oxlintrc(true, config.clone())
-                .expect("failed to build config")
-                .build()
-                .expect("failed to build config");
-            *linter = ServerLinter::new_with_linter(
-                Linter::new(LintOptions::default(), config_store).with_fix(FixKind::SafeFix),
+
+        let config_path = config_path?;
+        let oxlintrc = Oxlintrc::from_file(&config_path)
+            .expect("should have initialized linter with new options");
+        let config_store = ConfigStoreBuilder::from_oxlintrc(true, oxlintrc.clone())
+            .expect("failed to build config")
+            .build()
+            .expect("failed to build config");
+
+        #[expect(clippy::needless_late_init)]
```

Comment:
I couldn't figure out how to do this in a single line like `let linter = ...`. Not sure if it's something to do with using multiple lines in the if/else or what.

Reply:
Thanks, this got me what I needed here. Not sure what I was doing wrong, I tried a few things very similar to what you have here but I guess I must have missed something.